pandas/core/generic.py

```diff
@@ -1041,7 +1041,29 @@ def to_clipboard(self, excel=None, sep=None, **kwargs):
         from pandas.io import clipboard
         clipboard.to_clipboard(self, excel=excel, sep=sep, **kwargs)
 
-    #----------------------------------------------------------------------
+    def to_xarray(self):
+        """
+        Return an xarray object from the pandas object.
+
+        Returns
+        -------
+        a DataArray for a Series
+        a Dataset for a DataFrame
+        a Dataset for higher dims
+        """
+        import xray
+        if self.ndim == 1:
+            return xray.DataArray.from_series(self)
+        elif self.ndim == 2:
+            return xray.Dataset.from_dataframe(self)
```

Comment:
@shoyer should this be different from just `xray.DataArray(series)` or `xray.Dataset(df)`? 
This might be an xray change; seems a bit off to have special handling in pandas though